### PR TITLE
refactor "space" symbol to "whitespace"

### PR DESCRIPTION
**Pull request by amirali**
Hi!  
In the Regex, `\s` will match any whitespace, not just the space bar. So I think its symbol name is better to be `whitespace` instead of `space`.  
I also added `<space>` and `<alphabet>` symbols.  
And I'm sorry if it sounds like a dumb PR.

### CODE REVIEW

File: crates/melody_compiler/src/lib.rs [hunk 1]
```diff
@@ -62,8 +62,8 @@ enum Token {
     #[token("<digit>")]
     DigitSymbol,
 
-    #[token("<space>")]
-    SpaceSymbol,
+    #[token("<whitespace>")]
+    WhiteSpaceSymbol,
```
Comment (yoav-lavi) [inline crates/melody_compiler/src/lib.rs:66]:
```suggestion
    WhitespaceSymbol,
```

File: crates/melody_compiler/src/lib.rs [hunk 2]
```diff
@@ -274,7 +274,7 @@ pub fn compiler(source: &str) -> Result<String, ParseError> {
             // direct replacements
             Token::LineStart => handle_quantifier(String::from("^"), quantifier.clone(), false),
             Token::LineEnd => handle_quantifier(String::from("$"), quantifier.clone(), false),
-            Token::SpaceSymbol => handle_quantifier(String::from("\\s"), quantifier.clone(), false),
+            Token::WhiteSpaceSymbol => handle_quantifier(String::from("\\s"), quantifier.clone(), false),
```
Comment (yoav-lavi) [inline crates/melody_compiler/src/lib.rs:277]:
```suggestion
            Token::WhitespaceSymbol => handle_quantifier(String::from("\\s"), quantifier.clone(), false),
```

File: docs/docs/status.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ sidebar_position: 7
 | `capture { ... }`                   | `(...)`               | ✅          |
 | `capture name { ... }`              | `(?<name>...)`        | ✅          |
 | `match { ... }`                     | `(?:...)`             | ✅          |
-| `<space>;`                          | `\s`                  | ✅          |
+| `<whitespace>;`                          | `\s`                  | ✅          |
```
Comment (yoav-lavi) [inline docs/docs/status.md:24]:
```suggestion
| `<whitespace>;`                     | `\s`                  | ✅          |
```

File: docs/docs/status.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ sidebar_position: 7
 | Rust crate                          |                       | ✅          |
 | enforce group close                 |                       | 🐣          |
 | tests                               |                       | 🐣          |
-| `not <space>;`                      | `\S`                  | ❌          |
+| `not <whitespace>;`                      | `\S`                  | ❌          |
```
Comment (yoav-lavi) [inline docs/docs/status.md:54]:
```suggestion
| `not <whitespace>;`                 | `\S`                  | ❌          |
```

**Review by yoav-lavi — CHANGES_REQUESTED** [review]
A few minor tweaks

**Comment by yoav-lavi** [comment]
Not at all, there's actually an open issue for this!

https://github.com/yoav-lavi/melody/issues/32

**Comment by yoav-lavi** [comment]
We'd also need to change the examples in the readme / docs to fit this (possibly using " ")

**Comment by yoav-lavi** [comment]
Thank you for your contribution!

File: crates/melody_compiler/src/lib.rs (hunk 1 above)
Comment (amirali) [reply crates/melody_compiler/src/lib.rs]: 931ec457da9fc50a4f9ea185f67c30ece403f7ba

File: crates/melody_compiler/src/lib.rs (hunk 2 above)
Comment (amirali) [reply crates/melody_compiler/src/lib.rs]: 931ec457da9fc50a4f9ea185f67c30ece403f7ba

File: docs/docs/status.md (hunk 4 above)
Comment (amirali) [reply docs/docs/status.md]: 384e946e43a1eff0fee1ad5dc52750fce36ea093

**Comment by amirali** [comment]
And I also added the `<alphabet>` symbol (9dd17ae771093122540201081a04ebed05be3153) and `<space>` symbol (69a9439d5a2aa4728aac7d55e4a9d3a37a2b37f3)

**Comment by yoav-lavi** [comment]
Hey @amirali, could you please update this PR to the current state of the repo? Thank you!

**Comment by amirali** [comment]
Sorry I didn't notice it has conflict with main branch

**Comment by yoav-lavi** [comment]
No worries! That happened after the PR

**Review by yoav-lavi — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**